File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        size = len(nums)
        if size == 1:
            return 0;
        end = size - 1
        curCov, lastJump = 0, 0
        jumps = 0
        for i in range(size):
            curCov = max(curCov, i + nums[i])
            if i == lastJump:
                lastJump = curCov
                jumps += 1
                if curCov >= end:
                    return jumps
        return jumps
        

File: 0045-jump-game-ii/0045-jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    const size = nums.length;
    if (size === 1) {
        return 0;
    }
    let end = size - 1
    let curCoverage = 0
    let lastJump = 0;
    let jumps = 0;
    for (let i = 0; i < size; i++) {
        curCoverage = Math.max(curCoverage, i + nums[i]);
        if (i === lastJump) {
            lastJump = curCoverage;
            jumps++;
            if (curCoverage >= end) {
                return jumps;
            }
        }
    }
    return jumps;
};